### DISCUSSION TITLE
Displaying multiple images

### DISCUSSION THREAD

**Discussion by hirose125** [cbouy/bokehmol#2] [Ideas]
Is there a way to display all images contained within a selected range on a graph?

**Comment by cbouy** [comment]
Mmmh not that I know of.

There are some selection tools in bokeh, like the BoxSelectTool, but they don't trigger showing the tooltips by default, although the behavior could possibly be modified by specifying a custom callback on that tool. You could ask for some help on how to do that (trigger hover tooltips after a selection with BoxSelectTool) on the bokeh [community discourse](https://discourse.bokeh.org/c/community-support/9) or stackoverflow.

Another option is to specify `mode='vline'` or `mode='hline'` when you create the hover tool to trigger all tooltips on a vertical or horizontal line of where your cursor is. Not exactly what you're asking for though

**Reply by hirose125** [reply]
Thank you for your reply.

I'm currently satisfied with the functionalities, but I had a request to visualize and compare multiple molecules at the same time, so I posted in the discussion. As you mentioned, if we can get all the point information obtained with BoxSelectTool as a callback, I think we can achieve the function. I would like to consider implementing based on this source code.
And, I am a heavy user of mols2grid. Thank you for the amazing tool.

**Reply by cbouy** [reply]
Feel free to give it a go, as I said you can probably ask for some pointers on the bokeh [community support](https://discourse.bokeh.org/c/community-support/9) forum if you want to display the images directly in bokeh. The implementation will require mostly JavaScript knowledge, and then you can make a callback on the Python side following this guide on [CustomJS](https://docs.bokeh.org/en/3.2.0/docs/user_guide/interaction/js_callbacks.html#customjs-callbacks)

Or a simpler version would be to just get the indices that were selected by a tool for a `ColumnDataSource` with `source.selected.indices` and then subset the input pandas dataframe with `df.iloc[indices]` and simply display that with mols2grid in a separate notebook cell... Or even make a [python callback](https://docs.bokeh.org/en/3.2.1/docs/user_guide/interaction/python_callbacks.html#on-change-callback-triggers) with `source.selected.on_change(your_callback_function)`

And I'm glad to hear that you enjoy mols2grid, comments like these are really appreciated, thank you! 🙌
